test/unit/math/rev/mat/functor/gradient_test.cpp

```diff
@@ -46,6 +46,7 @@ TEST(AgradAutoDiff, gradient_threaded) {
   EXPECT_FLOAT_EQ(x_ref(0) * x_ref(0) + 3 * 2 * x_ref(1), grad_fx_ref(1));
 
   auto thread_job = [&](double x1, double x2) {
+    stan::math::ChainableStack thread_instance;
```

Comment:
Is this required now when using threads?

Reply:
Yes, you need an instance of ChainableStack in every thread. Otherwise the AD tape is not initialized. You can have more than one of those instances, but only the first one created is relevant.